src/login/pam_systemd.c

```diff
@@ -549,6 +549,26 @@ static bool getenv_harder_bool(pam_handle_t *handle, const char *key, bool fallb
         return r;
 }
 
+static uint32_t getenv_harder_uint32(pam_handle_t *handle, const char *key, uint32_t fallback) {
+        int r;
+
+        assert(handle);
+        assert(key);
+
+        const char *v = getenv_harder(handle, key, NULL);
+        if (isempty(v))
+                return fallback;
+
+        uint32_t u;
+        r = safe_atou32(v, &u);
+        if (r < 0) {
+                pam_syslog(handle, LOG_ERR, "Unsigned integer environment variable value of '%s' is not valid: %s", key, v);
```

Comment:
Maybe, better to log the fallback value? Also, we usually log in LOG_WARNING for errors that will be ignored.
`"Failed to parse environment variable value '%s' of '%s' as an unsigned integer, falling back to use %"PRIu32"."`